applications/minotari_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -1123,6 +1123,12 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
                 })?
                 .as_u64(),
         );
+
+        // assume full coinbase reward in case of single coinbase with zero value
+        if coinbases.len() == 1 && coinbases[0].value == 0 {
+            coinbases[0].value = reward as u64;
```

Comment:
```suggestion
            coinbases[0].value = 1;
```

This field is badly named. It should be `number_of_shares`. Simply setting it to 1 will get the full reward.

Reply:
Oh, I see. So technically this PR is only needed for GetNewBlockWithCoinbases. GetNewBlockTemplateWithCoinbases should work with value 1 like you suggested. I am currently using GetNewBlockTemplate/GetNewBlockWithCoinbases way (where it is almost impossible to find out correct coinbase value) but plan to transition to GetNewBlockTemplateWithCoinbases implementation, so probably this PR is not needed at all. Up to you.